src/xcode/ENA/ENA/Source/Scenes/Events/Store/MockEventStore.swift

```diff
@@ -0,0 +1,179 @@
+		checkins.removeAll { $0.id == id }
+		checkins.append(updatedCheckin)
+		checkinsPublisher.send(checkins)
+		return .success(())
+	}
+
+	func updateCheckin(id: Int, targetCheckinEndDate: Date) -> EventStoring.VoidResult {
+		var checkins = checkinsPublisher.value
+		guard let checkin = (checkins.first { $0.id == id }) else {
+			return .failure(.database(.unknown))
+		}
+		let updatedCheckin = makeUpdatedCheckin(with: checkin, targetCheckinEndDate: targetCheckinEndDate)
+		checkins.removeAll { $0.id == id }
+		checkins.append(updatedCheckin)
+		checkinsPublisher.send(checkins)
+		return .success(())
+	}
+
+	func deleteCheckin(id: Int) -> EventStoring.VoidResult {
+		checkinsPublisher.value.removeAll { $0.id == id }
+		return .success(())
+	}
+
+	func deleteAllCheckins() -> EventStoring.VoidResult {
+		checkinsPublisher.value = ([])
+		return .success(())
+	}
+
+	func createTraceTimeIntervalMatch(_ match: TraceTimeIntervalMatch) -> EventStoring.IdResult {
+		let id = UUID().hashValue
+		let traceTimeIntervalMatch = makeUpdatedTraceTimeIntervalMatch(with: match, id: id)
+		traceTimeIntervalMatchesPublisher.value.append(traceTimeIntervalMatch)
+		return .success(id)
+	}
+
+	func deleteTraceTimeIntervalMatch(id: Int) -> EventStoring.VoidResult {
+		traceTimeIntervalMatchesPublisher.value.removeAll { $0.id == id }
+		return .success(())
+	}
+
+	func createTraceWarningPackageMetadata(_ metadata: TraceWarningPackageMetadata) -> EventStoring.IdResult {
+		let id = UUID().hashValue
+		let traceWarningPackageMetadata = makeUpdatedTraceWarningPackageMetadata(with: metadata, id: id)
+		traceWarningPackageMetadatasPublisher.value.append(traceWarningPackageMetadata)
+		return .success(id)
+	}
+
+	func deleteTraceWarningPackageMetadata(id: Int) -> EventStoring.VoidResult {
+		traceWarningPackageMetadatasPublisher.value.removeAll { $0.id == id }
+		return .success(())
+	}
+
+	// MARK: - Protocol EventProviding
+
+	var traceLocationsPublisher = CurrentValueSubject<[TraceLocation], Never>([])
+
+	var checkinsPublisher = CurrentValueSubject<[Checkin], Never>([])
+
+	var traceTimeIntervalMatchesPublisher = OpenCombine.CurrentValueSubject<[TraceTimeIntervalMatch], Never>([])
+
+	var traceWarningPackageMetadatasPublisher = OpenCombine.CurrentValueSubject<[TraceWarningPackageMetadata], Never>([])
+
+	// MARK: - Private
+
+	private func makeUpdatedTraceTimeIntervalMatch(with match: TraceTimeIntervalMatch, id: Int) -> TraceTimeIntervalMatch {
+		TraceTimeIntervalMatch(
+			id: match.id,
+			checkinId: match.checkinId,
+			traceWarningPackageId: match.traceWarningPackageId,
+			traceLocationGUID: match.traceLocationGUID,
+			transmissionRiskLevel: match.transmissionRiskLevel,
+			startIntervalNumber: match.startIntervalNumber,
+			endIntervalNumber: match.endIntervalNumber
+		)
+	}
+
+	private func makeUpdatedTraceWarningPackageMetadata(with metadata: TraceWarningPackageMetadata, id: Int) -> TraceWarningPackageMetadata {
```

Comment:
Just a nice-to-have idea: These could be extensions, so one could write `traceWarningPackageMetadata.updatedWith(id: 17)` or something similar, returning a new instance

Reply:
Could be a private extension just here in the mock store

Reply:
👍Done